src/v/raft/consensus.cc

```diff
@@ -3529,4 +3533,77 @@ std::optional<uint8_t> consensus::get_under_replicated() const {
     return count;
 }
 
+reply_result consensus::lightweight_heartbeat(
+  model::node_id source_node, model::node_id target_node) {
+    /**
+     * Handle lightweight heartbeat
+     */
+    if (unlikely(target_node != _self.id())) {
+        vlog(
+          _ctxlog.warn,
+          "received lw_heartbeat request addressed to different node: {}, "
+          "current node: {}, source: {}",
+          target_node,
+          _self,
+          source_node);
+        return reply_result::failure;
+    }
+
+    _hbeat = clock_type::now();
+    return reply_result::success;
```

Comment:
I wonder if there are cases where we need to reply with a full reply here? Something may have changed on the follower as well that we need to notify the leader of, right?

Reply:
i think i will introduce a new `reply_result` for this purpose that will force reseting heartbeat state on leader